java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/SSL/SSLEngine.java

```diff
@@ -182,37 +151,29 @@ public void initialize() {
                     password = null;
                     keyManagers = kmf.getKeyManagers();
 
-                    //
                     // If no alias is specified, we look for the first key entry in the key store.
                     //
-                    // This is required to force the key manager to always choose a certificate
-                    // even if there's no certificate signed by any of the CA names sent by the
-                    // server. Ice servers might indeed not always send the CA names of their
-                    // trusted roots.
-                    //
+                    // This is required to force the key manager to always choose a certificate even if there's no
+                    // certificate signed by any of the CA names sent by the server. Ice servers might indeed not
+                    // always send the CA names of their trusted roots.
                     if (alias.isEmpty()) {
-                        for (Enumeration<String> e = keys.aliases();
-                                e.hasMoreElements(); ) {
+                        for (Enumeration<String> e = keys.aliases(); e.hasMoreElements(); ) {
                             String a = e.nextElement();
                             if (keys.isKeyEntry(a)) {
                                 alias = a;
                                 break;
                             }
                         }
-                    }
-
-                    if (!alias.isEmpty()) {
-                        //
-                        // If the user selected a specific alias, we need to wrap the key managers
-                        // in order to return the desired alias.
-                        //
+                    } else {
+                        // If the user selected a specific alias, ensure it correspond with a key entry.
                         if (!keys.isKeyEntry(alias)) {
```

Comment:
Previously we check this below in ` if (!alias.isEmpty()) {` but is only useful here, when the "alias" was provided by the user. Don't need to do that when we just retrieve it from the keystore doing the same check.